File: sampleCode/R/Services/Extensions.R
setClass("Extensions",
         slots = list(
           dummy = "character"
         ))

# Create a generic function for Create
setGeneric("ThrowIfNull", function(object,value,valueName = deparse(substitute(object))) standardGeneric("ThrowIfNull"))

# Define a method for the Create function with the Projects class
setMethod("ThrowIfNull", signature = "Extensions", function(object,value,valueName = deparse(substitute(object))){
  
  if (!is.null(value)) {
    return(value)
  }
  warning(paste("ArgumentNullException:", valueName))
  return(NULL)
  
})

extension <- new("Extensions")

File: sampleCode/Python/services/extensions.py
from typing import TypeVar, Optional

T = TypeVar('T')

class ArgumentNullException(Exception):
    pass

class Extensions:
    @staticmethod
    def throw_if_null(value: Optional[T], value_name: Optional[str] = None) -> T:
        if value is not None:
            return value
        raise ArgumentNullException(f'Argument "{value_name}" is null.')

    @staticmethod
    def is_null_or_whitespace(s: Optional[str]) -> bool:
        return s is None or s.strip() == ''


